habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -139,6 +139,140 @@ def get_ao_global_bb(
     return cumulative_global_bb
 
 
+def get_bb_for_object_id(
+    sim: habitat_sim.Simulator,
+    obj_id: int,
+    ao_link_map: Dict[int, int] = None,
+    ao_aabbs: Dict[int, mn.Range3D] = None,
+) -> Tuple[mn.Range3D, mn.Matrix4]:
+    """
+    Wrapper to get a bb and global transform directly from an object id.
+    Handles RigidObject and ArticulatedLink ids.
+
+    :param sim: The Simulator instance.
+    :param obj_id: The integer id of the object or link.
+    :param ao_link_map: A pre-computed map from link object ids to their parent ArticulatedObject's object id.
+    :param ao_aabbs: A pre-computed map from ArticulatedObject object_ids to their local bounding boxes. If not provided, recomputed as necessary.
+
+    :return: tuple (local_aabb, global_transform)
+    """
+
+    # stage bounding box
+    if obj_id == habitat_sim.stage_id:
+        return (
+            sim.get_active_scene_graph().get_root_node().cumulative_bb,
+            mn.Matrix4.identity_init(),
+        )
+
+    obj = get_obj_from_id(sim, obj_id, ao_link_map)
+
+    assert (
+        obj is not None
+    ), f"object id {obj_id} is not found, this is unexpected. Invalid/stale object id?"
```

Comment:
Should `raise` an error instead of calling `assert`. Assert gets skipped outside of debug mode.